src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1390 @@
+            init_weight(module.tok_embeddings, stds["embedding"])
+        elif isinstance(module, ModernBertMLP):
+            init_weight(module.Wi, stds["in"])
+            init_weight(module.Wo, stds["out"])
+        elif isinstance(module, ModernBertAttention):
+            init_weight(module.Wqkv, stds["in"])
+            init_weight(module.Wo, stds["out"])
+        elif isinstance(module, ModernBertPredictionHead):
+            init_weight(module.dense, stds["in"])  # TODO: Should this be "out"/"final_out"?
+        elif isinstance(module, ModernBertPoolingHead):
+            init_weight(module.dense, stds["out"])
+        elif isinstance(module, ModernBertForMaskedLM):
+            init_weight(module.decoder, stds["out"])
+        elif isinstance(module, (ModernBertForSequenceClassification, ModernBertForTokenClassification)):
+            init_weight(module.classifier, stds["final_out"])
+
+    @torch.no_grad()
+    def _unpad_inputs_no_grad(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: torch.Tensor,
+        position_ids: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+    ):
+        return self._unpad_inputs(
+            input_ids=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+        )
+
+    def _unpad_inputs(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: torch.Tensor,
+        position_ids: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+    ) -> Tuple[torch.Tensor | int | None]:
+        return _unpad_modernbert_input(
+            inputs=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+        )
+
+    @torch.no_grad()
+    def _pad_outputs_no_grad(
+        self,
+        inputs: torch.Tensor,
+        indices: torch.Tensor,
+        batch_size: int,
+        seqlen: int,
+        labels: Optional[torch.Tensor] = None,
+        ignore_index: int = -100,
+    ):
+        return self._pad_outputs(
+            inputs=inputs,
+            indices=indices,
+            batch_size=batch_size,
+            seqlen=seqlen,
+            labels=labels,
+            ignore_index=ignore_index,
+        )
+
+    def _pad_outputs(
+        self,
+        inputs: torch.Tensor,
+        indices: torch.Tensor,
+        batch_size: int,
+        seqlen: int,
+        labels: Optional[torch.Tensor] = None,
+        ignore_index: int = -100,
+    ):
+        return _pad_modernbert_output(
+            inputs=inputs, indices=indices, batch=batch_size, seqlen=seqlen, labels=labels, ignore_index=ignore_index
+        )
+
+    @classmethod
+    def offsets_to_sequence_ids_tensor(cls, offsets):
+        device = offsets.device
+        counts = offsets[1:] - offsets[:-1]
+        return torch.repeat_interleave(torch.arange(len(counts), device=device, dtype=torch.int32), counts)
+
+    @torch.compile(dynamic=False)
```

Comment:
@staghado do we need this compile? We now have a `compile` argument that we need to abide by. Ideally I'd remove this compile unless it's necessary for matching the research performance or if it helps speed a lot.

Otherwise we can't get Flex Attention on e.g. Windows or MPS (currently we can't either way, due to the FA2 dependency for Flex).

Reply:
we can set it according to the compile argument or remove it for now. We can add it later when it's more stable

Reply:
I've blanket removed it here: 1e367df9b9e0a92189c562445d42d09b859c22d1
In the other cases with `torch.compile`, we have to duplicate all the code in the function, which was a bit much here.